### PR TITLE
EvmTool fixes

### PR DESCRIPTION
**Pull request by shemnon**
## PR description

A mixed collection of EVMTool fixes
* Ensure ECDSA algo is pre-configured
* used clamped math in an old gas calculator
* use correct gas calculator in fluent APIs

### CODE REVIEW

File: evm/src/main/java/org/hyperledger/besu/evm/MainnetEVMs.java [hunk 1]
```diff
@@ -546,7 +547,7 @@ public static EVM berlin(final EvmConfiguration evmConfiguration) {
    * @return the evm
    */
   public static EVM berlin(final BigInteger chainId, final EvmConfiguration evmConfiguration) {
-    return berlin(new IstanbulGasCalculator(), chainId, evmConfiguration);
+    return berlin(new BerlinGasCalculator(), chainId, evmConfiguration);
```
Comment (macfarla) [inline evm/src/main/java/org/hyperledger/besu/evm/MainnetEVMs.java:550]: this makes sense, just wondering if changing this is going to break anything? it's been this way for a long time!

**Review by macfarla — COMMENTED** [review]
LGTM just one Q on the MainnetEVMs change

File: evm/src/main/java/org/hyperledger/besu/evm/MainnetEVMs.java (hunk 1 above)
Comment (shemnon) [reply evm/src/main/java/org/hyperledger/besu/evm/MainnetEVMs.java:550]: It's broken and wrong.  Should have caught it earlier.  It doesn't align with what the real definition of berlin is.  Also, only impacts users of the relatively new fluent evm APIs.  One researcher found a differential fuzzing bug based off of it, and this was the bug.

Comment (macfarla) [reply evm/src/main/java/org/hyperledger/besu/evm/MainnetEVMs.java:550]: ok makes sense then

**Review by macfarla — APPROVED** [review]

**Pull request merged**